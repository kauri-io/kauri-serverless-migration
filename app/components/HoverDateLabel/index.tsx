import React from 'react';
import { Label } from "../Typography";
import moment from 'moment';

interface IDate {
  date: string;
  hovered?: boolean;
  status: undefined | "PUBLISHED" | "DRAFT";
}

class Date extends React.Component<IDate,IDate> {
    constructor(props: IDate) {
      super(props);
      this.state = {
        date: props.date,
        hovered: false,
        status: props.status,
      }
    }
  
    public render() {
      return (
        <div onMouseEnter={() => this.setState({ hovered: true})} onMouseLeave={() => this.setState({ hovered: false})}>
          <Label>
            {this.state.hovered ?
              (this.props.status === "DRAFT" ? "Drafted " : "Posted ") + moment(this.props.date).format('DD MMM YYYY HH:mm') :
              (this.props.status === "DRAFT" ? "Drafted " : "Posted ") + moment(this.props.date).fromNow()}
          </Label>
        </div>
      );
    }
  }
export default Date;