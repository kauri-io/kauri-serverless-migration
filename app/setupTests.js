import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
