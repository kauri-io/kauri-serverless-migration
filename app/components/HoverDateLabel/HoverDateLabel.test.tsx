import HoverDateLabel, { HoverDateStatus } from './index'
import { mountWithTheme } from '../../setupTests'

let props = {
    date: '2019-01-01',
    status: 'PUBLISHED' as HoverDateStatus,
}

describe('components/HoverDateLabel', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(<HoverDateLabel {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
