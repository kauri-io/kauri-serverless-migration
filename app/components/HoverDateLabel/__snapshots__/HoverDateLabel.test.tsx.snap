// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HoverDateLabel snapshot should match 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <Date
    date="2019-01-01"
    status="PUBLISHED"
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <Typography__component>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Typography__component-sc-36xkf6-0",
                "isStatic": false,
                "lastClassName": "kALGmD",
                "rules": Array [
                  "font-size:11px;font-weight:bold;text-transform:uppercase;color:",
                  [Function],
                  ";text-align:",
                  [Function],
                  ";",
                  [Function],
                  " ",
                  [Function],
                ],
              },
              "displayName": "Typography__component",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Typography__component-sc-36xkf6-0",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <span
            className="Typography__component-sc-36xkf6-0 kALGmD"
          >
            Posted 7 months ago
          </span>
        </StyledComponent>
      </Typography__component>
    </div>
  </Date>
</ThemeProvider>
`;
